examples/systemcalls/systemcalls.c

```diff
@@ -58,7 +124,24 @@ bool do_exec(int count, ...)
  *   as second argument to the execv() command.
  *
 */
-
+    
+    int status;
+    pid_t pid;
+    pid=fork();
+    if(pid==-1)
+    	return false;
+    else if(pid == 0)
+    {
+    	if((execv(full_path, rest_of_args))!=-1)
```

Comment:
@abaa3023 I'm not sure the logic is right here - I would expect to see `execv(command[0], command)`  see https://linux.die.net/man/3/execv

> The first argument, by convention, should point to the filename associated with the file being executed.

Reply:
Also there are other error conditions besides -1, < 0 would be a better check here.